casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala

```diff
@@ -194,10 +195,11 @@ class MultiParentCasperImpl[F[_]: Sync: Concurrent: Capture: ConnectionsCell: Tr
       b: BlockMessage
   ): F[Boolean] =
     for {
-      blockStoreContains <- BlockStore[F].contains(b.blockHash)
-      state              <- Cell[F, CasperState].read
-      bufferContains     = state.blockBuffer.exists(_.blockHash == b.blockHash)
-    } yield (blockStoreContains || bufferContains)
+      dag            <- blockDag
+      dagContains    <- dag.contains(b.blockHash)
+      state          <- Cell[F, CasperState].read
+      bufferContains = state.blockBuffer.exists(_ == b.blockHash)
```

Comment:
You could replace this with `state.blockBuffer.contains(b.blockHash)`.

Reply:
👍